lib/nepali_calendar/fiscal_year.rb

```diff
@@ -0,0 +1,90 @@
+class NepaliCalendar::FiscalYear
+
+    attr_accessor :start_year, :end_year
+
+    def initialize(start_year, end_year)
+        @start_year = start_year
+        @end_year = end_year
+    end
+        
+        # Should return_the_date in bs
+    def beginning_of_year
+        start_date = start_year.to_s.prepend('20')
+        NepaliCalendar::Calendar.new(nil, {year: start_date, month: 4, day: 1})
+    end
+        
+        # Should return the date in bs
+    def end_of_year
+        end_date = end_year.to_s.prepend('20')
+        NepaliCalendar::Calendar.new(nil, {year: end_date, month: 3, day: NepaliCalendar::BS[end_date.to_i][3]})
+    end
+
+      # Returns the bs date that is the beginning of the provided fiscal_year.
+    # def beginning_of_fiscal_year_in_bs
+    #     beginning_of_year #start date of fiscal year
+    # end
+
+
+      # Returns the bs date that is the end of the provided fiscal_year.
+    # def self.end_of_fiscal_year_in_bs(fiscal_year)
+    #     start_year = fiscal_year.to_s.slice(0, 2)
+    #     end_year = fiscal_year.to_s.slice(2,2)
+    #     FiscalYear.new(start_year, end_year).end_of_year
+    # end
+
+    def self.get_fiscal_year_from_ad(ad_date)
+        bs_date = NepaliCalendar::BsCalendar.ad_to_bs(ad_date.year.to_s, ad_date.month.to_s, ad_date.day.to_s)
+        if bs_date.month < 4
+          fiscal_year = ((ad_date.year - 1).to_s.slice(2,2)).to_s + ad_date.year.to_s.slice(2,2).to_s
+        else
+          fiscal_year =ad_date.year.to_s.slice(2,2).to_s + ((ad_date.year + 1).to_s.slice(2,2)).to_s
+        end
+    end
+
+    def self.fiscal_year_for_bs_date(bs_year, bs_month, bs_day) # (2079, 2, 12) ==> 7879
```

Comment:
Let's not separately support the passing of the year, month, and day. Instead, let's have it take one parameter, which should be the 'NepaliCalendar::Calendar' object.